File: android-agent/src/main/java/io/opentelemetry/android/internal/services/visiblescreen/activities/Pre29VisibleScreenLifecycleBinding.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.internal.services.visiblescreen.activities;

import android.app.Activity;
import androidx.annotation.NonNull;
import io.opentelemetry.android.internal.services.visiblescreen.VisibleScreenService;

/**
 * An ActivityLifecycleCallbacks that is responsible for telling the VisibleScreenTracker when an
 * activity has been resumed and when an activity has been paused. It's just a glue class designed
 * for API level before 29.
 */
public class Pre29VisibleScreenLifecycleBinding implements DefaultingActivityLifecycleCallbacks {

    private final VisibleScreenService visibleScreenService;

    public Pre29VisibleScreenLifecycleBinding(VisibleScreenService visibleScreenService) {
        this.visibleScreenService = visibleScreenService;
    }

    @Override
    public void onActivityResumed(@NonNull Activity activity) {
        visibleScreenService.activityResumed(activity);
    }

    @Override
    public void onActivityPaused(@NonNull Activity activity) {
        visibleScreenService.activityPaused(activity);
    }
}

File: services/src/main/java/io/opentelemetry/android/internal/services/visiblescreen/activities/Pre29VisibleScreenLifecycleBinding.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.internal.services.visiblescreen.activities

import android.app.Activity
import io.opentelemetry.android.internal.services.visiblescreen.VisibleScreenTracker

/**
 * An ActivityLifecycleCallbacks that is responsible for telling the VisibleScreenTracker when an
 * activity has been resumed and when an activity has been paused. It's just a glue class designed
 * for API level before 29.
 */
class Pre29VisibleScreenLifecycleBinding(
    private val visibleScreenTracker: VisibleScreenTracker,
) : DefaultingActivityLifecycleCallbacks {
    override fun onActivityResumed(activity: Activity) {
        visibleScreenTracker.activityResumed(activity)
    }

    override fun onActivityPaused(activity: Activity) {
        visibleScreenTracker.activityPaused(activity)
    }
}
